src/ethereum-json-rpc-client/src/lib.rs

```diff
@@ -1,4 +1,286 @@
+
+        let mut results = Vec::new();
+
+        let value_from_json = |value| serde_json::from_value::<R>(value);
+
+        // Collect chunks before iteration, otherwise the future won't be `Send`
+        let chunks = params
+            .into_iter()
+            .chunks(max_batch_size)
+            .into_iter()
+            .map(Iterator::collect::<Vec<_>>)
+            .collect::<Vec<_>>();
+        for chunk in chunks {
+            let method_calls = chunk
+                .into_iter()
+                .map(|(params, id)| {
+                    Call::MethodCall(MethodCall {
+                        jsonrpc: Some(Version::V2),
+                        method: method.clone(),
+                        params,
+                        id,
+                    })
+                })
+                .collect::<Vec<_>>();
+            let chunk_size = method_calls.len();
+            let request = Request::Batch(method_calls);
+
+            let response = if is_update_call {
+                self.client.send_rpc_update_request(request).await?
+            } else {
+                self.client.send_rpc_query_request(request).await?
+            };
+
+            match response {
+                Response::Single(response) => match response {
+                    Output::Success(result) => {
+                        if chunk_size == 1 {
+                            results.push(value_from_json(result.result)?);
+                        } else {
+                            anyhow::bail!(
+                                "unexpected number of results: have: 1, expected {chunk_size}"
+                            );
+                        }
+                    }
+                    Output::Failure(err) => {
+                        anyhow::bail!("{err:?}");
+                    }
+                },
+                Response::Batch(response) => {
+                    if chunk_size == response.len() {
+                        for resp in response.into_iter() {
+                            match resp {
+                                Output::Success(resp) => {
+                                    results.push(value_from_json(resp.result)?)
+                                }
+                                Output::Failure(err) => {
+                                    anyhow::bail!("{err:?}");
+                                }
+                            }
+                        }
+                    } else {
+                        anyhow::bail!(
+                            "unexpected number of results: have: {}, expected {}",
+                            response.len(),
+                            chunk_size
+                        );
+                    }
+                }
+            }
+        }
+
+        Ok(results)
+    }
+}
+
+#[inline]
+fn is_update_call(method: &str) -> bool {
+    method.eq(ETH_SEND_RAW_TRANSACTION_METHOD)
+}
+
+pub trait Client: Clone + Send + Sync {
```

Comment:
```suggestion
pub trait Client: Clone + Send {
```

Can we avoid Sync here? Isn't Clone + Send enough?

Reply:
If I remove `Sync` I have a compilation error in the evm-archiver with `tokio:spawn`. I'll double check it

Reply:
I think we can try to replace `&client` with `client.clone()` in such places.